Source/Plugins/Core/com.equella.core/js/tsrc/legacycontent/LegacyContent.tsx

```diff
@@ -220,9 +237,19 @@ export const LegacyContent = React.memo(function LegacyContent({
           }
         }
       }
-      const form = document.getElementById("eqpageForm") as HTMLFormElement;
-      const vals = collectParams(form, command, [].slice.call(arguments, 1));
-      submitCurrentForm(true, false, form.action, vals);
+      const form = getEqPageForm();
+      if (!form) {
+        onError(
+          generateFromError({
+            name: "stdSubmit Failure",
+            message:
+              "stdSubmit unable to proceed due to missing " + legacyFormId,
+          })
+        );
+      } else {
+        const vals = collectParams(form, command, [].slice.call(arguments, 1));
```

Comment:
destructuring may be a tad easier to read here, thoughts?

```suggestion
        const [_, ...args] = arguments;
        const vals = collectParams(form, command, args);
```

Reply:
Ha, nice pick-up. When I relocated this code, the complexity did catch my attention, but I thought I'd leave as is in case it wasn't doing something 'special'. So I didn't stop and actually read what it was doing.

There is a bunch of code in here which still falls in that category, and I may come back to after all is working.

Reply:
Yeah, nah, I'm not touching this bit. Why is it using the magical `arguments` variable anyway rather than just placing that in the function signature... Oh, probably because the callback function being returned is being called by the legacy AJAX code... Gah, I'm leaving this be on that premise alone.